airflow/contrib/operators/jenkins_job_trigger_operator.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.jenkins_hook import JenkinsHook
+import jenkins
+
+try:
+    basestring
+except NameError:
+    basestring = str # For python3 compatibility
+
+
+class JenkinsJobTriggerOperator(BaseOperator):
+    """
+    Trigger a Jenkins Job and monitor it's execution
+    :param jenkins_connection_id: The jenkins connection to use for this job
+    :type jenkins_connection_id: string
+    :param job_name: The name of the job to trigger
+    :type job_name: string
+    :param parameters: The parameters block to provide to jenkins
+    :type parameters: string
+    :param sleep_time: How long will the operator sleep between each status request for the job (min 1, default 10)
+    :type sleep_time: int
+    :param max_try_before_job_appears: The maximum number of requests to make \
+        while waiting for the job to appears on jenkins server (default 10)
+    :type max_try_before_job_appears: int
+    """
+    template_fields = ('parameters',)
+    template_ext = ('.json',)
+    ui_color = '#f9ec86'
+
+    @apply_defaults
+    def __init__(self,
+                 jenkins_connection_id="",
+                 job_name="",
```

Comment:
Is empty name acceptable?

Reply:
I raise an exception after is the name is empty. Should I remove the default value to force the user to give a name ?

Reply:
Default argument removed, I hope the tests will still pass :D